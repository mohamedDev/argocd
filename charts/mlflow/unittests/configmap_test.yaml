suite: test configmap

templates:
  - configmap.yaml

release:
  name: mlflow
  namespace: mlflow

chart:
  version: 1.0.0
  appVersion: 1.0.0

tests:
  - it: should have postgres environments when postgres is enabled
    set:
      backendStore.postgres.enabled: true
      backendStore.postgres.host: postgres-service
      backendStore.postgres.port: 5432
      backendStore.postgres.database: postgres
    asserts:
      - hasDocuments:
          count: 1
      - equal:
          path: data.PGHOST
          value: postgres-service
      - equal:
          path: data.PGPORT
          value: "5432"
      - equal:
          path: data.PGDATABASE
          value: postgres

  - it: should have postgres environments and migration configmaps when postgres and migration are enabled
    set:
      backendStore.postgres.enabled: true
      backendStore.postgres.host: postgres-service
      backendStore.postgres.port: 5432
      backendStore.postgres.database: postgres
      backendStore.databaseMigration: true
    asserts:
      - hasDocuments:
          count: 2

  - it: should have postgres environments and database connection checker configmaps when postgres and database connection check are enabled
    set:
      backendStore.postgres.enabled: true
      backendStore.postgres.host: postgres-service
      backendStore.postgres.port: 5432
      backendStore.postgres.database: postgres
      backendStore.databaseConnectionCheck: true
    asserts:
      - hasDocuments:
          count: 2

  - it: should have mysql environments when mysql is enabled
    set:
      backendStore.mysql.enabled: true
      backendStore.mysql.host: mysql-service
      backendStore.mysql.port: 3306
      backendStore.mysql.database: mysql
    asserts:
      - hasDocuments:
          count: 1
      - equal:
          path: data.MYSQL_HOST
          value: mysql-service
      - equal:
          path: data.MYSQL_TCP_PORT
          value: "3306"
      - equal:
          path: data.MYSQL_DATABASE
          value: mysql

  - it: should have mysql environments when bitnami mysql is enabled
    set:
      mysql:
        enabled: true
        auth:
          database: custom-db
    asserts:
      - hasDocuments:
          count: 1
      - equal:
          path: data.MYSQL_HOST
          value: mlflow-mysql
      - equal:
          path: data.MYSQL_TCP_PORT
          value: "3306"
      - equal:
          path: data.MYSQL_DATABASE
          value: custom-db

  - it: should have mysql environments and migration configmaps when mysql and migration are enabled
    set:
      backendStore.mysql.enabled: true
      backendStore.mysql.host: mysql-service
      backendStore.mysql.port: 3306
      backendStore.mysql.database: mysql
      backendStore.databaseMigration: true
    asserts:
      - hasDocuments:
          count: 2

  - it: should have mysql environments and migration configmaps when bitnamimysql and migration are enabled
    set:
      mysql:
        enabled: true
        auth:
          database: custom-db
      backendStore.databaseMigration: true
    asserts:
      - hasDocuments:
          count: 2

  - it: should have postgresql environments when postgresql is enabled
    set:
      postgresql:
        enabled: true
        auth:
          database: custom-db
    documentSelector:
      path: metadata.name
      value: mlflow-env-configmap
    asserts:
      - equal:
          path: data.PGHOST
          value: mlflow-postgresql
      - equal:
          path: data.PGPORT
          value: "5432"
      - equal:
          path: data.PGDATABASE
          value: custom-db

  - it: should use different postgresql port when postgresql is enabled and postgresql.primary.service.ports.postgresql is set
    set:
      postgresql:
        enabled: true
        primary:
          service:
            ports:
              postgresql: 5433
    documentSelector:
      path: metadata.name
      value: mlflow-env-configmap
    asserts:
      - equal:
          path: data.PGPORT
          value: "5433"

  - it: should have postgresql environments and migration configmaps when postgres and migration are enabled
    set:
      postgresql:
        enabled: true
      backendStore:
        databaseMigration: true
    asserts:
      - hasDocuments:
          count: 2

  - it: should have postgresql environments and database connection checker configmaps when postgres and database connection check are enabled
    set:
      postgresql:
        enabled: true
      backendStore:
        databaseConnectionCheck: true
    asserts:
      - hasDocuments:
          count: 2

  - it: should have log level configuration when log level is set
    set:
      log:
        enabled: true
        level: debug
    asserts:
      - equal:
          path: data.MLFLOW_CONFIGURE_LOGGING
          value: "true"
      - equal:
          path: data.MLFLOW_LOGGING_LEVEL
          value: DEBUG

  - it: should disable telemetry on default
    asserts:
      - equal:
          path: data.MLFLOW_DISABLE_TELEMETRY
          value: "true"
      - equal:
          path: data.DO_NOT_TRACK
          value: "true"

  - it: should match snapshot of default values
    release:
      name: mlflow
      namespace: mlflow
    chart:
      version: 1.0.0
      appVersion: 1.0.0
    asserts:
      - matchSnapshot: { }

  - it: should match snapshot of default values with postgres
    set:
      backendStore:
        postgres:
          enabled: true
          host: postgres-custom-service
          port: 5432
          database: mlflow
          user: mlflowuser
          password: mlflowpassword
    release:
      name: mlflow
      namespace: mlflow
    chart:
      version: 1.0.0
      appVersion: 1.0.0
    asserts:
      - matchSnapshot: { }

  - it: should match snapshot of default values with mysql
    set:
      backendStore:
        mysql:
          enabled: true
          host: mysql-custom-service
          port: 3306
          database: mlflow
          user: mlflowuser
          password: mlflowpassword
    release:
      name: mlflow
      namespace: mlflow
    chart:
      version: 1.0.0
      appVersion: 1.0.0
    asserts:
      - matchSnapshot: { }

  - it: should match snapshot of default values with bitnami postgresql
    set:
      postgresql:
        enabled: true
    release:
      name: mlflow
      namespace: mlflow
    chart:
      version: 1.0.0
      appVersion: 1.0.0
    asserts:
      - matchSnapshot: { }

  - it: should match snapshot of default values with bitnami mysql
    set:
      mysql:
        enabled: true
    release:
      name: mlflow
      namespace: mlflow
    chart:
      version: 1.0.0
      appVersion: 1.0.0
    asserts:
      - matchSnapshot: { }
