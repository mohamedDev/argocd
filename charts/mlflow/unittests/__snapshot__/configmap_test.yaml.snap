should match snapshot of default values:
  1: |
    apiVersion: v1
    data:
      DO_NOT_TRACK: "true"
      MLFLOW_CONFIGURE_LOGGING: "true"
      MLFLOW_DISABLE_TELEMETRY: "true"
      MLFLOW_LOGGING_LEVEL: INFO
    kind: ConfigMap
    metadata:
      labels:
        app: mlflow
        chart: mlflow-1.0.0
        heritage: Helm
        release: mlflow
      name: mlflow-env-configmap
should match snapshot of default values with bitnami mysql:
  1: |
    apiVersion: v1
    data:
      DO_NOT_TRACK: "true"
      MLFLOW_CONFIGURE_LOGGING: "true"
      MLFLOW_DISABLE_TELEMETRY: "true"
      MLFLOW_LOGGING_LEVEL: INFO
      MYSQL_DATABASE: mlflow
      MYSQL_HOST: mlflow-mysql
      MYSQL_TCP_PORT: "3306"
    kind: ConfigMap
    metadata:
      labels:
        app: mlflow
        chart: mlflow-1.0.0
        heritage: Helm
        release: mlflow
      name: mlflow-env-configmap
should match snapshot of default values with bitnami postgresql:
  1: |
    apiVersion: v1
    data:
      DO_NOT_TRACK: "true"
      MLFLOW_CONFIGURE_LOGGING: "true"
      MLFLOW_DISABLE_TELEMETRY: "true"
      MLFLOW_LOGGING_LEVEL: INFO
      PGDATABASE: mlflow
      PGHOST: mlflow-postgresql
      PGPORT: "5432"
    kind: ConfigMap
    metadata:
      labels:
        app: mlflow
        chart: mlflow-1.0.0
        heritage: Helm
        release: mlflow
      name: mlflow-env-configmap
should match snapshot of default values with mysql:
  1: |
    apiVersion: v1
    data:
      DO_NOT_TRACK: "true"
      MLFLOW_CONFIGURE_LOGGING: "true"
      MLFLOW_DISABLE_TELEMETRY: "true"
      MLFLOW_LOGGING_LEVEL: INFO
      MYSQL_DATABASE: mlflow
      MYSQL_HOST: mysql-custom-service
      MYSQL_TCP_PORT: "3306"
    kind: ConfigMap
    metadata:
      labels:
        app: mlflow
        chart: mlflow-1.0.0
        heritage: Helm
        release: mlflow
      name: mlflow-env-configmap
should match snapshot of default values with postgres:
  1: |
    apiVersion: v1
    data:
      DO_NOT_TRACK: "true"
      MLFLOW_CONFIGURE_LOGGING: "true"
      MLFLOW_DISABLE_TELEMETRY: "true"
      MLFLOW_LOGGING_LEVEL: INFO
      PGDATABASE: mlflow
      PGHOST: postgres-custom-service
      PGPORT: "5432"
    kind: ConfigMap
    metadata:
      labels:
        app: mlflow
        chart: mlflow-1.0.0
        heritage: Helm
        release: mlflow
      name: mlflow-env-configmap
